.github/workflows/build.yml

```
@@ -154,9 +152,134 @@ jobs:
         name: haskell-language-server-wrapper-${{ runner.OS }}${{env.EXE_EXT}}.${{ steps.compress_wrapper_binary.outputs.extension }}
         path: ${{ steps.compress_wrapper_binary.outputs.path }}
 
+  build-linux:
+    runs-on: ubuntu-latest
+    container: alpine:3.12
```

could not this be set conditionally to not have to duplicate the entire job?

---

Do you know how?

---

```suggestion
    container: ${{ (matrix.os == 'ubuntu-18.04' && 'alpine:3.12') || '' }}
```
https://github.com/haskell/haskell-language-server/compare/alpine-check-build...jneira:alpine-check-build?expand=1

this trick seems to work, i.e. it starts ubuntu in a container and the rest without a container, but the build within the container is failing due to other errors: 
https://github.com/jneira/haskell-language-server/actions/runs/1564975136

---

And the fail seems to be related with the install of ghc using ghcup and apt-get:

``` [ Info  ] downloading: https://raw.githubusercontent.com/haskell/ghcup-metadata/master/ghcup-0.0.6.yaml as file /github/home/.ghcup/cache/ghcup-0.0.6.yaml
  ghcup: curl: executeFile: does not exist (No such file or directory)
  [ Warn  ] Could not get download info, trying cached version (this may not be recent!)
  [ ...   ] If this problem persists, consider switching downloader via: 
  [ ...   ]     ghcup config set downloader Wget
  [ Error ] JSON decoding failed with: YAML exception:
  [ ...   ] Yaml file not found: /github/home/.ghcup/cache/ghcup-0.0.6.yaml
  [ ...   ] Consider removing /github/home/.ghcup/cache/ghcup-0.0.6.yaml manually.
  [ ...   ] 
```

and there is no sudo in alpine it seems:

```
Error: Unable to locate executable file: sudo. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
```

---

https://stackoverflow.com/questions/49225976/use-sudo-inside-dockerfile-alpine

---

i think we should try to use the same image than gitlab:

https://github.com/haskell/haskell-language-server/blob/779439876855872c523088e400526c6f1807bca2/.gitlab-ci.yml#L127-L129

---

```suggestion
    container: ${{ (matrix.os == 'ubuntu-18.04' && 'docker://registry.gitlab.haskell.org/ghc/ci-images/x86_64-linux-alpine3_12:4ed1a4f27828ba96a34662dc954335e29b470cd2') || '' }}
```

this get the gitlab image

https://github.com/jneira/haskell-language-server/runs/4488270256?check_suite_focus=true

but it has to be tuned to make the rest of the job work: install curl and whatever else, alpine seems to be a minimalistic image

---

> but it has to be tuned to make the rest of the job work: install curl and whatever else, alpine seems to be a minimalistic image

oh you already was doing it here: https://github.com/haskell/haskell-language-server/pull/2463/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R168

---

The gitlab image throws a permission error trying to do the git checkout. It has cabal/ghc preinstalled, not sure what would be other changes compared with the base alpine:3.12